lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+  # Helper
+  $app->helper(cache => sub { state $cache = {} });
+
+  # Application
+  $app->cache->{foo} = 'bar';
+  my $result = $app->cache->{foo};
+
+  # Controller
+  $c->cache->{foo} = 'bar';
+  my $result = $c->cache->{foo};
+
+  # Template
+  % cache->{foo} = 'bar';
+  %= cache->{foo}
+
+=head2 hook
+
+  $app->hook(after_dispatch => sub {...});
+
+Extend L<Mojolicious> with hooks, which allow code to be shared with all requests indiscriminately, for a full list of
+available hooks see L</"HOOKS">.
+
+  # Dispatchers will not run if there's already a response code defined
+  $app->hook(before_dispatch => sub ($c) {
+    $c->render(text => 'Skipped static file server and router!')
+      if $c->req->url->path->to_route =~ /do_not_dispatch/;
+  });
+
+=head2 new
+
+  my $app = Mojolicious->new;
+  my $app = Mojolicious->new(moniker => 'foo_bar');
+  my $app = Mojolicious->new({moniker => 'foo_bar'});
+
+Construct a new L<Mojolicious> application and call L</"startup">. Will automatically detect your home directory. Also
+sets up the renderer, static file server, a default set of plugins and an L</"around_dispatch"> hook with the default
+exception handling.
+
+=head2 plugin
+
+  $app->plugin('some_thing');
+  $app->plugin('some_thing', foo => 23);
+  $app->plugin('some_thing', {foo => 23});
+  $app->plugin('SomeThing');
+  $app->plugin('SomeThing', foo => 23);
+  $app->plugin('SomeThing', {foo => 23});
+  $app->plugin('MyApp::Plugin::SomeThing');
+  $app->plugin('MyApp::Plugin::SomeThing', foo => 23);
+  $app->plugin('MyApp::Plugin::SomeThing', {foo => 23});
+
+Load a plugin, for a full list of example plugins included in the L<Mojolicious> distribution see
+L<Mojolicious::Plugins/"PLUGINS">.
+
+=head2 server
+
+  $app->server(Mojo::Server->new);
+
+Emits the L</"before_server_start"> hook.
+
+=head2 start
+
+  $app->start;
+  $app->start(@ARGV);
+
+Start the command line interface for your application. For a full list of commands that are available by default see
+L<Mojolicious::Commands/"COMMANDS">. Note that the options C<-h>/C<--help>, C<--home> and C<-m>/C<--mode>, which are
+shared by all commands, will be parsed from C<@ARGV> during compile time.
+
+  # Always start daemon
+  $app->start('daemon', '-l', 'http://*:8080');
+
+=head2 startup
+
+  $app->startup;
+
+This is your main hook into the application, it will be called at application startup. Meant to be overloaded in a
+subclass.
+
+  sub startup ($self) {...}
+
+=head2 warmup
+
+  $app->warmup;
+
+Preload classes from L</"preload_namespaces"> for future use.
+
+=head1 HELPERS
+
+In addition to the L</"ATTRIBUTES"> and L</"METHODS"> above you can also call helpers on L<Mojolicious> objects. This
+includes all helpers from L<Mojolicious::Plugin::DefaultHelpers> and L<Mojolicious::Plugin::TagHelpers>. Note that
+application helpers are always called with a new default controller object, so they can't depend on or change
+controller state, which includes request, response and stash.
```

Comment:
Is there a way to know or distinguish that a given helper is intended to be used on Mojolicious objects versus controller objects?